### PR TITLE
Source-related classes refactor

### PR DESCRIPTION
**Pull request by VictorVelarde**
This PR includes several refactors related to 'Source' classes (MVT, GeoJSON...)

Refactors:
- [x] Refactor inside `MVT` class into smaller, more meaningful functions
- [x] Small refactor in `BaseWindshaft` class
- [x] Refactor in Windshaft client + new `WindshaftRequestHelper` class
- [x] Refactor GeoJSON class:
    - [x] Internal reorg into smaller functions
    - [x] Extract GeoJSONGeometryTransformer for coords transformations
    - [x] Extract GeoJSONMetadataBuilder for metadata related operations

### CODE REVIEW

File: src/client/WindshaftRequestHelper.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import { version } from '../../package';
+import CartoMapsAPIError, { CartoMapsAPIErrorTypes } from '../errors/carto-maps-api-error';
+
+const REQUEST_GET_MAX_URL_LENGTH = 2048;
+
+export default class WinshaftRequestHelper {
```
Comment (elenatorro) [inline src/client/WindshaftRequestHelper.js]: `WinshaftRequestHelper` -> `WindshaftRequestHelper`

File: src/client/WindshaftRequestHelper.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    constructor (conf, mapConfig) {
+        this._conf = conf;
+        this._mapConfig = mapConfig;
+
+        this._auth = this._encodeParameter('api_key', this._conf.apiKey);
+        this._client = this._encodeParameter('client', `vl-${version}`);
+    }
+
+    async getMapRequest () {
```
Comment (elenatorro) [inline src/client/WindshaftRequestHelper.js]: I'd name it `getLayerGroup` cause we're returning a layergroup, not a request.

File: src/client/windshaft.js [hunk 3]
```diff
@@ -303,48 +345,32 @@ export default class Windshaft {
...
+            sample: {
+                num_rows: SAMPLE_ROWS,
+                include_columns: includedColumns // TODO: when supported by Maps API: exclude_columns: excludedColumns
+            }
         };
     }
 
+    async _getLayerGroupFromWindshaft (conf, mapConfigAgg) {
+        const requestHelper = new WindshaftRequestHelper(conf, mapConfigAgg);
+        const layergroup = await requestHelper.getMapRequest();
```
Comment (elenatorro) [inline src/client/windshaft.js]: If we change `getMapRequest` to `getLayerGroup` we can just return it:

```js
return await requestHelper.getLayerGroup();
```

File: src/client/windshaft.js [hunk 4]
```diff
@@ -107,13 +130,15 @@ export default class Windshaft {
...
-        return !schema.equals(this._MNS, MNS) ||
-            resolution !== this.resolution ||
-            (
-                JSON.stringify(filtering) !== JSON.stringify(this.filtering) &&
-                this.metadata.featureCount > MIN_FILTERING
-            );
+    _needToInstantiateMap (vizInfo) {
+        const { MNS, resolution, filtering } = vizInfo;
+
+        const schemaChanged = (schema.notEquals(this._MNS, MNS));
```
Comment (elenatorro) [inline src/client/windshaft.js]:
```js
const schemaChanged = schema.notEquals(this._MNS, MNS);
```

File: src/client/windshaft.js [hunk 5]
```diff
@@ -122,19 +147,20 @@ export default class Windshaft {
...
-            // default choices
-            backendFilters: true
+            backendFilters: true // default choices
         };
-        if (metadata) {
-            if (metadata.featureCount >= 0) {
-                choices.backendFilters = metadata.featureCount > MIN_FILTERING || !metadata.backendFiltersApplied;
-            }
+
+        if (metadata && metadata.featureCount >= 0) {
```
Comment (elenatorro) [inline src/client/windshaft.js]: Nice one! maybe just:

```js
const backendFilters = metadata && metadata.featureCount >= 0
    ? metadata.featureCount > MIN_FILTERING || !metadata.backendFiltersApplied
    : true

return { backendFilters }
```

File: src/client/windshaft.js [hunk 6]
```diff
@@ -261,7 +289,7 @@ export default class Windshaft {
     }
 
     _buildSelectClause (MNS) {
-        const columns = Object.keys(MNS).concat(['the_geom_webmercator', 'cartodb_id']);
+        const columns = Object.keys(MNS).concat(['the_geom_webmercator', ID_PROPERTY]);
```
Comment (elenatorro) [inline src/client/windshaft.js]: I love having `cartodb_id` as a constant ✨ we've defined it twice as `ID_PROPERTY` and `DEFAULT_ID_PROPERTY`. As a minor improvement I'd keep one of those, but it's just a suggestion, it's ok like this!

File: src/sources/MVT.js [hunk 7]
```diff
@@ -86,38 +112,63 @@ export default class MVT extends Base {
...
-                });
-                this._metadataSent = true;
-                this._workerDispatch[this._mID] = resolve;
-                this._mID++;
-            });
-        },
-        zoom => this._options.maxZoom === undefined
-            ? this._options.viewportZoomToSourceZoom(zoom)
-            : Math.min(this._options.viewportZoomToSourceZoom(zoom), this._options.maxZoom)
+        const urlToDataframeTransformer = this._urlToDataframeTransformer.bind(this);
```
Comment (elenatorro) [inline src/sources/MVT.js]: ❤️

**Comment by elenatorro** [comment]
BTW, I think we should include this in the CHANGELOG

File: src/client/WindshaftRequestHelper.js (hunk 2 above)
Comment (VictorVelarde) [reply src/client/WindshaftRequestHelper.js]: Totally 👍

File: src/client/windshaft.js (hunk 5 above)
Comment (VictorVelarde) [reply src/client/windshaft.js]: Thx! I'm gonna add a new explaining variable, but I think I'll keep the simple IF, to keep the current `shouldBeServerFiltered` variable

File: src/client/windshaft.js (hunk 6 above)
Comment (VictorVelarde) [reply src/client/windshaft.js]: Nice!, I'll keep just DEFAULT_ID_PROPERTY const from Metadata

**Comment by VictorVelarde** [comment]
Thx @elenatorro! I have made some small adjustments as suggested by your CR

**Comment by VictorVelarde** [comment]
I have already merge this into 'develop', for the next release

**Pull request merged**